Updating Frontend Project (#16)
IZEA/interview: front_end/README.md

# Ember Project

> We've put your soft skills to the test, now we want to see some dope code! For this part of the interview, we would like you to create a small Ember.js app using this [API](https://reqres.in/). We would like for you to use ember-data to talk to this API, but if that turns out too complex, feel free to use a simpler library like [ember-ajax](https://github.com/ember-cli/ember-ajax).

Reviewer: Maybe:
```suggestion
We've put your soft skills to the test, now we want to see some dope code! For this part of the interview, we would like you to create a small Ember.js app that requests and shows data from the [Reqres API](https://reqres.in/). We would like for you to use ember-data to talk to this API, but if you're having trouble getting ember-data to work, you can use a simpler addon like [ember-ajax](https://github.com/ember-cli/ember-ajax).
```

If any special requirements have been conveyed to you from your interview
team, they supersede the following.

## Requirements

> - [ ] Create a paginated list of users.

Reviewer: Maybe:
```suggestion
- [ ] Display a paginated list of users. The data should be retrieved from the [Reqres API](https://reqres.in/), and there should be pagination controls to retrieve additional pages of data.
```

> - [ ] Each user should be able to toggle a side sliding panel/modal that is populated with that user's data.

Reviewer: Maybe:
```suggestion
- [ ] Each user should be able to toggle a side sliding panel/modal that is populated with that user's data. You can make this look & behave pretty much however you want, but it should not trigger a full route transition.
```

- [ ] Style it to the best of your ability using sass.

> - [ ] Do not use a front end framework such as bootstrap.

Reviewer: Should we clarify that it's fine to use a utility framework like tailwind or neat?

Author: That's fine if you think it's important. Should I mention making it as responsive as possible too?

Reviewer: Yeah, that makes sense!

- [ ] Each user should be able to toggle a side sliding panel/modal that is populated with that user's data.
- [ ] Style it to the best of your ability using sass.
- [ ] Do not use a front end framework such as bootstrap.
- [ ] You can choose to either include the filler text as written, or provide your own.
- [ ] Use the images provided in the assets directory where applicable.
- [ ] The page should be responsive rather than have a separate mobile version.
- [ ] Aside from the provided images, no other images should be used in the creation of your page.
- [ ] If you do not have access to the Helvetica font, you may substitute with a sans-serif of your choosing.
- [ ] Hover state for navigation links should use the same style as their active state (As represented in mock images).
- [ ] Visited state for navigation links should use the same style as their inactive state (As represented in mock images).

> ## Bonus Points

Reviewer: What do you think about leaving the Bonus Points section and specify something about showing loading states?

Reviewer: Also, what about testing? Should that be a requirement or a bonus points item?

Author: That's a good idea. I was thinking we should keep it with something but I wasn't sure what to have there. Maybe also the use of sass variables? Just a thought.

Reviewer: Yeah, thinking about it more, I think testing should be a hard requirement for this. I think maybe we can expand on it in the Bonus Points section, though. So having any tests that prove your functionality works at all should satisfy the requirement, but bonus points for using [ember-test-selectors](https://github.com/simplabs/ember-test-selectors) to isolate how you select elements in tests from your styling code and [ember-cli-mirage](https://www.ember-cli-mirage.com/) to isolate your tests from talking to a real API on the internet.

If any special requirements have been conveyed to you from your interview
team, they supersede the following.

## Requirements

-- [ ] Display a paginated list of posts. The data should be retrieved from the [JSONPlaceholder API](https://jsonplaceholder.typicode.com/), and there should be pagination controls to retrieve additional pages of data.
-- [ ] Each post should be able to toggle a side sliding panel/modal that is populated with that user's data. You can make this look & behave pretty much however you want, but it should not trigger a full route transition.
- [ ] We are not providing a mock for this, you will need to figure out a design that you feel represents the best user experience.
- [ ] Do not use a front end framework such as bootstrap, but it's fine to use a utility framework like tailwind or neat.
- [ ] The UI should be fairly responsive.
- [ ] Write tests that prove your functionality works.

## Bonus Points

> - [ ] Use [ember-cli-sass](https://github.com/aexmachina/ember-cli-sass)

Reviewer: ```suggestion
- [ ] Use [ember-cli-sass](https://github.com/aexmachina/ember-cli-sass) instead of vanilla CSS to style your app.
```
